internal/backup_new.go

```diff
@@ -42,39 +59,41 @@ func checkDbDirectoryForUnwrapNew(dbDataDirectory string, sentinelDto BackupSent
 // TODO : unit tests
 // Do the job of unpacking Backup object
 func (backup *Backup) unwrapNew(
-	dbDataDirectory string, sentinelDto BackupSentinelDto, filesToUnwrap map[string]bool, createIncrementalFiles bool,
-) error {
+	dbDataDirectory string, sentinelDto BackupSentinelDto, filesToUnwrap map[string]bool,
+	createIncrementalFiles bool) (*UnwrapResult, error) {
 	useNewUnwrapImplementation = true
 	err := checkDbDirectoryForUnwrapNew(dbDataDirectory, sentinelDto)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	tarInterpreter := NewFileTarInterpreter(dbDataDirectory, sentinelDto, filesToUnwrap, createIncrementalFiles)
 	tarsToExtract, pgControlKey, err := backup.getTarsToExtract(sentinelDto, filesToUnwrap)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// Check name for backwards compatibility. Will check for `pg_control` if WALG version of backup.
 	needPgControl := IsPgControlRequired(backup, sentinelDto)
 
 	if pgControlKey == "" && needPgControl {
-		return newPgControlNotFoundError()
+		return nil, newPgControlNotFoundError()
 	}
 
 	err = ExtractAll(tarInterpreter, tarsToExtract)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if needPgControl {
 		err = ExtractAll(tarInterpreter, []ReaderMaker{newStorageReaderMaker(backup.getTarPartitionFolder(), pgControlKey)})
 		if err != nil {
-			return errors.Wrap(err, "failed to extract pg_control")
+			return nil, errors.Wrap(err, "failed to extract pg_control")
 		}
 	}
 
 	tracelog.InfoLogger.Print("\nBackup extraction complete.\n")
-	return nil
+	unwrapResult := newUnwrapResult(tarInterpreter.CompletedFiles, tarInterpreter.CreatedPageFiles,
```

Comment:
Don't repeat yourself. You have the same fields in tarInterpreter and in unwrapResult.

Reply:
Fixed in 3880c5da3d1ab954ef87a0bc380b27e9a3992fce